client/src/ui/search.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  CheckIcon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+  SearchIcon,
+} from "@liftedinit/ui";
+import { FormEvent, useState } from "react";
+import { NavigateFunction, useNavigate } from "react-router-dom";
+
+const validTerms = [
+  { path: "blocks", regex: /^\d+$/ },
+  { path: "transactions", regex: /^[0-9a-f]{64}$/ },
+  { path: "address", regex: /^m[0-9a-zA-Z]{49,54}$/ },
```

Comment:
Would love your eyes on these, @hansl . We could obviously get more fancy (block height has to be less than current height, etc.) but does anything jump out as totally wrong?